Test interrupt fails intermittently

We're seeing this error pop up from time to time on travis.ci.  cc @jcrist 

```python
________________________________ test_interrupt ________________________________
[gw0] linux -- Python 3.5.3 /home/travis/miniconda/envs/test-environment/bin/python
@pytest.mark.slow
    def test_interrupt():
        try:
            proc = subprocess.Popen([sys.executable, '-c', code],
                                    stdout=subprocess.PIPE,
                                    stderr=subprocess.PIPE)
            # Wait for scheduler to start
            msg = proc.stdout.readline()
            if msg != 'started\n' and proc.poll() is not None:
                assert False, "subprocess failed"
            # Scheduler has started, send an interrupt
            sigint = signal.CTRL_C_EVENT if os.name == 'nt' else signal.SIGINT
            try:
                proc.send_signal(sigint)
                # Wait a bit for it to die
                start = time()
                while time() - start < 2.0:
                    if proc.poll() is not None:
                        break
                    sleep(0.05)
                else:
>                   assert False, "KeyboardInterrupt Failed"
E                   AssertionError: KeyboardInterrupt Failed
E                   assert False
dask/tests/test_threaded.py:155: AssertionError
```

--- a/dask/tests/test_threaded.py
+++ b/dask/tests/test_threaded.py
@@ -107,6 +107,7 @@ def test_thread_safety():
 
 code = """
 import sys
+import time
 from dask.threaded import get
 
 def signal_started():
@@ -114,11 +115,7 @@ def signal_started():
     sys.stdout.flush()
 
 def long_task(x):
-    out = 0
-    N = 100000
-    for i in range(N):
-        for j in range(N):
-            out += 1
+    time.sleep(5)
 
 dsk = {('x', i): (long_task, 'started') for i in range(100)}
 dsk['started'] = (signal_started,)
@@ -147,7 +144,7 @@ def test_interrupt():
             proc.send_signal(sigint)
             # Wait a bit for it to die
             start = time()
-            while time() - start < 2.0:
+            while time() - start < 4.0:
                 if proc.poll() is not None:
                     break
                 sleep(0.05)

